distributed/shuffle/multi_comm.py

```diff
@@ -142,7 +142,7 @@ async def communicate(self):
                 await self.queue.get()
 
             with self.lock:
-                address = max(self.sizes, key=self.sizes.get)
+                address = max(self.sizes, key=self.sizes.get)  # type: ignore
```

Comment:
```suggestion
                address = max(self.sizes, key=self.sizes.__getitem__)
```

Note there's an extra bug here in that if `self.sizes` is empty, this raises ValueError.

Reply:
Thanks. A bit confuse but this code is only reached if it is non-empty. `sizes` and `shards` are coupled and this loop checks this in the beginning and `continue`s until this is filled.